cli/api/api.go

```diff
@@ -135,6 +140,7 @@ func (a *api) connectMaster() (master.ClientInterface, error) {
 		if err != nil {
 			return nil, fmt.Errorf("could not create a client to the master: %s", err)
 		}
+		a.authenticateHost()
```

Comment:
Well, kind of a catch-22. It's expected to fail in the case of `serviced host add ...` for adding the master host because there's no key file to start with, and add-host will work without it.